Kickstarter-iOS/Views/ProjectStatesContainerView.swift

```diff
@@ -0,0 +1,88 @@
+import Library
+import KsApi
+import Prelude
+import UIKit
+
+class ProjectStatesContainerView: UIView {
+  fileprivate let vm: ProjectStatesContainerViewViewModelType = ProjectStatesContainerViewViewModel()
+  // MARK: - Properties
+
+  private lazy var rootStackView: UIStackView = { UIStackView(frame: .zero) }()
```

Comment:
If we're not doing any additional configuration, I'd say we should prefer using straight `let`s:

`private let rootStackView = UIStackView(frame: .zero)`

What do others think about this? cc @justinswart @dusi

Reply:
I did it this way because it was previously suggested I should. This is a convention that the team should figure out because it's getting confusing. cc @justinswart @dusi

Reply:
I also thought we agreed on `lazy var`s for the time being...most importantly because as `lazy vars` support both single and multi line assignments compared to `let`s that don't.

For example

```swift
lazy var tableView: UITableView = { UITableView(frame: .zero, style: .plain) }
```

for single line property

```swift
lazy var tableView: UITableView = { 
  UITableView(frame: .zero, style: .plain)
    |> \.dataSource .~ self
}
```

for multilane property (the above is not supported by Swift - as far as I know)

in addition to that `lazy var`s are loaded lazily so if there's a view component we need to add later in the lifecycle (as oppose to init) this will be optimized solution.

Reply:
Not all properties require access to `self` to be configured. Furthermore, I'd argue that we should always be explicitly setting things like `delegate`s and `dataSource`s in `viewDidLoad` so that we have certainty on when things those are getting assigned - otherwise, it becomes harder to reason about exactly when (for example) the table view's `dataSource` is getting set, since it's lazily evaluated 🤷‍♀ 

I guess my feeling is that we should only be using `lazy vars` when we actually need to defer a property's initialization for some reason, for example: it's only needed in certain cases, or it's very expensive to initialize (date formatters etc). Most properties we're looking to create are needed almost right away (when they're added to the view hierarchy, usually on `viewDidLoad`) which I *think* makes their deferred initialization mostly pointless. My feeling is that *always* using `lazy vars` to declare properties even when their "lazy" functionality isn't really needed is mostly a stylistic preference. If we want to use that approach just so everything's consistent, then I guess that's fine.